crates/ty_server/src/server/api/diagnostics.rs

```diff
@@ -139,10 +147,6 @@ pub(super) fn clear_diagnostics(session: &Session, uri: &lsp_types::Url, client:
 ///
 /// [publish diagnostics notification]: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_publishDiagnostics
 pub(super) fn publish_diagnostics(session: &Session, url: &lsp_types::Url, client: &Client) {
```

Comment:
I think we could rename this to `publish_diagnostics_if_needed` to be consistent with the `clear_diagnostics_*` version.